cli/mfg.c

```diff
@@ -404,12 +412,87 @@ static int image_select(int argc, char **argv)
 	return ret;
 }
 
+static int secure_state_set(int argc, char **argv)
+{
+	int ret;
+	enum switchtec_boot_phase phase_id;
+	struct switchtec_security_cfg_stat state = {};
+
+	const char *desc = "Set device secure state (BL1 and Main Firmware only)\n\n"
+			   "This command is can only be used when device "
+			   "secure state is UNINITIALIZED_UNSECURED.";
```

Comment:
Definitely need more here. What does this actually do? Why would anyone care about this flag?

I'm also worried about a trend that all these new commands are just thin wrappers around basic MRPC commands. You need to consider what your users want to do or get and the use cases, not just provide thin wrappers for everything the hardware does. 

Without understanding what these commands are doing it's hard for me to review anything, but the sense I'm getting is that there should be a higher level interface. Certainly you shouldn't be setting a secure state without actually initializing some kind of security feature... Changing a flag on the device is pretty useless.

Reply:
Added more detailed descriptions, please check if sufficient. @lsgunth

Reply:
Some background information (excerpts from command descriptions):

A device can be in one of these three secure states:
UNINITIALIZED_UNSECURED: the default state when chip is shipped. All
security-related settings are **uninitialized**.
INITIALIZED_UNSECURED: this is the state when security-related settings are
'initialized', and the chip is set to **unsecured** state.
INITIALIZED_SECURED: this is the state when security-related settings are
'initialized', and the chip is set to **secured** state.

When device is shipped, it comes in _uninitialized-unsecured_ mode. From this mode, user can use 'config-set' and/or other mfg commands to initialize security settings. When user is satisfied with the settings, they use this command to switch chip to either INITIALIZED_SECURED or INITIALIZED_UNSECURED state, and the chip stays in that state till end of life. 

This command is the 'ENABLE' button for security-related settings.

We provide this (and a few other) commands as thin wrappers of basic MRPC commands to offer user better flexibility. As a usecase, user can program the configurations during manufacturing time, and before deployment, decide to enable security only on half of the units, leaving the other half in unsecured mode.